libraries/chain/apply_context.cpp

```diff
@@ -144,10 +144,12 @@ void apply_context::exec_one()
       r.auth_sequence[auth.actor] = next_auth_sequence( auth.actor );
    }
 
+   auto input_hash = digest_type::hash(act->data);
    if( control.is_builtin_activated( builtin_protocol_feature_t::action_return_value ) ) {
-      r.return_value.emplace( std::move( action_return_value ) );
+      auto output_hash = digest_type::hash(action_return_value);
```

Comment:
Both of these `digest_type::hash` should use instead `encode` method of `wasm_interface` which includes `checktime` calls.

Reply:
Forgot to add that.  Thanks

Reply:
Resolved.